nhaystack-rt/src/nhaystack/server/tags/BNSiteRefRelation.java

```diff
@@ -134,16 +146,4 @@ private static boolean hasSiteTag(Entity entity)
     {
         return entity.tags().contains(ID_SITE);
     }
-
-    private static Relation getRelation(BComponent component, Id id)
-    {
-        for (Relation relation : component.relations())

```

Comment:
This iterator requires a getAll relations that ends up evaluating all tag rules- very expensive.